File: automator/src/androidTest/java/com/stardust/automator/simple_action/DepthFirstSearchTargetActionTest.java
package com.stardust.automator.simple_action;

import android.view.accessibility.AccessibilityNodeInfo;

import com.stardust.automator.UiObject;
import com.stardust.automator.test.TestUiObject;

import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.*;

/**
 * Created by Stardust on 2017/5/5.
 */
public class DepthFirstSearchTargetActionTest {

    @Test
    public void perform() {
        DepthFirstSearchTargetAction action = new DepthFirstSearchTargetAction(AccessibilityNodeInfo.ACTION_CLICK, new FilterAction.Filter() {
            @Override
            public List<UiObject> filter(UiObject root) {
                List<UiObject> list = new ArrayList<>();
                for (int i = 0; i < root.getChildCount(); i++) {
                    list.add(root.child(i));
                }
                return list;
            }
        });
        TestUiObject root = new TestUiObject(5);
        action.perform(root);
        System.out.println(TestUiObject.max);
        assertEquals(1, TestUiObject.count);
    }

}

File: automator/src/androidTest/java/com/stardust/automator/simple_action/DepthFirstSearchTargetActionTest.kt
package com.stardust.automator.simple_action

import android.view.accessibility.AccessibilityNodeInfo

import com.stardust.automator.UiObject
import com.stardust.automator.test.TestUiObject

import org.junit.Test

import java.util.ArrayList

import org.junit.Assert.*

/**
 * Created by Stardust on 2017/5/5.
 */
class DepthFirstSearchTargetActionTest {

    @Test
    fun perform() {
        val action = DepthFirstSearchTargetAction(AccessibilityNodeInfo.ACTION_CLICK, FilterAction.Filter { root ->
            val list = ArrayList<UiObject>()
            for (i in 0 until root.childCount) {
                list.add(root.child(i))
            }
            list
        })
        val root = TestUiObject(5)
        action.perform(root)
        println(TestUiObject.max)
        assertEquals(1, TestUiObject.count.toLong())
    }

}